tests/integration/devfile/cmd_dev_test.go

```diff
@@ -203,6 +204,26 @@ var _ = Describe("odo dev command tests", func() {
 					Expect(err).ToNot(HaveOccurred())
 				})
 			})
+			When("recording telemetry data", func() {
+				BeforeEach(func() {
+					helper.EnableTelemetryDebug()
+					session, _, _, _, _ := helper.StartDevMode()
+					session.Stop()
+				})
+				AfterEach(func() {
+					helper.ResetTelemetry()
+				})
+				It("should record the telemetry data correctly", func() {
+					td := helper.GetTelemetryDebugData()
+					Expect(td.Event).To(ContainSubstring("odo dev"))
+					Expect(td.Properties.Success).To(BeFalse())
+					Expect(td.Properties.Error).To(ContainSubstring("interrupt"))
+					Expect(td.Properties.ErrorType == "*errors.errorString").To(BeTrue())
```

Comment:
In the case of `odo dev`, interrupt is not really interrupt that we want to catch as an error. Perhaps you can add a check for `odo dev` before the signal watcher starts.
https://github.com/redhat-developer/odo/blob/main/pkg/odo/genericclioptions/runnable.go#L73

Check if `cmd.Name() == "dev"`, and if it is, don't add the `os.Interrupt` signal to the list. @feloy what do you think?

Reply:
Should this not be its own issue?

Reply:
I guess, but since you were working on the telemetry for v3, I figured we could include it here. But you're right, I'll open a new issue for this, we'll probably need some discussion on this.

Reply:
This seems to be a much more complicated issue than it appears.
The interrupt being an error could entirely depend on where the interrupt happened as well like before the command to `wait for something to change` for eg